// @awesome-cordova-plugins/native-audio/index.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import { AwesomeCordovaNativePlugin, cordova } from '@awesome-cordova-plugins/core';
var NativeAudioOriginal = /** @class */ (function (_super) {
    __extends(NativeAudioOriginal, _super);
    function NativeAudioOriginal() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    NativeAudioOriginal.prototype.preloadSimple = function (id, assetPath) { return cordova(this, "preloadSimple", {}, arguments); };
    NativeAudioOriginal.prototype.preloadComplex = function (id, assetPath, volume, voices, delay) { return cordova(this, "preloadComplex", {}, arguments); };
    NativeAudioOriginal.prototype.play = function (id, completeCallback) { return cordova(this, "play", { "successIndex": 1, "errorIndex": 2 }, arguments); };
    NativeAudioOriginal.prototype.stop = function (id) { return cordova(this, "stop", {}, arguments); };
    NativeAudioOriginal.prototype.loop = function (id) { return cordova(this, "loop", {}, arguments); };
    NativeAudioOriginal.prototype.unload = function (id) { return cordova(this, "unload", {}, arguments); };
    NativeAudioOriginal.prototype.setVolumeForComplexAsset = function (id, volume) { return cordova(this, "setVolumeForComplexAsset", {}, arguments); };
    NativeAudioOriginal.pluginName = "NativeAudio";
    NativeAudioOriginal.plugin = "cordova-plugin-nativeaudio";
    NativeAudioOriginal.pluginRef = "plugins.NativeAudio";
    NativeAudioOriginal.repo = "https://github.com/floatinghotpot/cordova-plugin-nativeaudio";
    NativeAudioOriginal.platforms = ["Android", "Browser", "iOS"];
    return NativeAudioOriginal;
}(AwesomeCordovaNativePlugin));
var NativeAudio = new NativeAudioOriginal();
export { NativeAudio };
